classes/User/User.mjs

```diff
@@ -22,51 +22,102 @@ export class User {
   async getUserById() {
     // returns user's public info
     // find() will be replaced with getById() when the find error in DatabaseController() is fixed
-    let user = await database.find({
+
+    
+    if (!this.id) {
+      err_out.message = "No user ID found. Instantiate User class with a proper ID as follows new User(UserId)"
+      err_out.status = 400
+      throw err_out
+    }
+
+    return  database.find({
       _id: this.id,
       "@type": "User"
     })
-    user = user[0]
-    this.id = user?._id
-    const publicUser = includeOnly(user, "profile", "_id")
-    this.user = publicUser
-    return publicUser
+    .then(resp => {
+      if (resp instanceof Error) { 
+        throw resp;
```

Comment:
superfluous `;`, get rid of this one and the others like it in this file.  I also don't think it will make it here if the resp is an Error so you might not need this if at all (I am not sure 100% sure about this).